ldk/javascript/examples/consolidate-loop/listenHotkey.js

```diff
@@ -0,0 +1,154 @@
+                    }
+                },
+                  type: TextInput,
+                  },
+                  {
+                    label: 'Date of Birth',
+                    onChange: (error,value) => 
+                    {
+                      patient.setDob(value);
+                    },
+                    type: TextInput,
+                    tooltip: "MM/dd/yyyy",
+                  },
+                  {
+                    label: 'Gender',
+                    onChange: (error,value) => 
+                    {
+                        patient.setGender(value);
+                    },
+                    type: TextInput,
+                  },
+                  {
+                    label: 'Telephone',
+                    onChange: (error,value) => 
+                    {
+                        patient.setTelephone(value);
+                    },
+                    type: Telephone,
+                  },
+                  {
+                    label: 'Email',
+                    onChange: (error,value) =>
+                    {
+                        patient.setEmail(value);
+                    },
+                    type: Email,
+                  },
+                  {
+                    label: 'Visit Reason',
+                    onChange: (error,value) => 
+                    {
+                        patient.setVisitReason(value);
+                    },
+                    type: TextInput,
+                  },
+                  {
+                    label: 'Appointment Date',
+                    onChange: (error,value) => 
+                    {
+                        patient.setAppointmentDate(value);
+                    },
+                    type: TextInput,
+                  },
+                  {
+                    label: 'Appointment Time',
+                    onChange: (error,value) => 
+                    {
+                        patient.setAppointmentTime();
+                    },
+                    type: TextInput,
+                  },
+                  {
+                    // Close&clear the loop after clicking submit 
+                    label: 'Submit',
+                    onClick: async () => 
+                    {
+                      try {
+                        patient.verify()
+                      } catch (e) {
+                        console.error(e)
+                        return
+                      }
+                        // varify if having duplicated patients
+                        // If patients have same name and dob,use email to distinguish. 
+                      let cred1 = `${patient._firstName}:${patient._lastName}:${patient._dob}` 
+                      let cred2 = patient._email
+                      let patientRecord = '|' + patient.seralize()
+
+                      const found = await new Promise((resolve, reject) => {
+
+                        filesystem.readFile("./PatientInfo.txt").then((data) => {
```

Comment:
I would use await pattern here to reduce a mix of the .then/await to make it more readable.